Multisig README & folder/files restructuring

Imod7 opened the pull request:
## Description
- Added a detailed `README.md` file for the multisig example that describes : 
   - How to set up and run the `multisig` example 
   - The key takeaways
- Restructuring the files and folders under [txwrapper-examples](https://github.com/paritytech/txwrapper-core/tree/main/packages/txwrapper-examples) folder. 

## Proposed files and folders structure
- Under the [txwrapper-examples](https://github.com/paritytech/txwrapper-core/tree/main/packages/txwrapper-examples) folder there is a general `README.md` file that gives a short description about what is included in the current folder.
- Each example is in a separate folder, inside of which there are files and subfolders (eg. "src" and "images") related to the specific example. The folders/examples that we have now are the following :
  - polkadot
  - polkadotBatchAll
  - multisig
- Created a folder called `common` and moved file `util.ts` there since it is used in multiple examples.

## Pendings
- [x] Add a description in the general `README.md` file that gives a short description about what is included in the current folder.
- [x] Update the `README.md` file in the `polkadotBatchAll` folder.

Imod7 committed:
Multisig README & folder-files restructure
- Created a new folder 'docs' under the examples folder for all the README files.
- Moved the README file for Polkadot under the 'docs' folder and renamed it to 'POLKADOT.md'.
- Created a README file for the Multisig example called 'MULTISIG.md'.
- Added a folder 'images' in the 'docs' folder with screenshots from running the Multisig example.

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
```diff
@@ -0,0 +1,117 @@
+# Multisig example
+
+## Description
+The multisig example code shows how to create a Multisig account and execute a transaction from this newly created account to send some funds to another address.
```
```suggestion
The multisig example code shows how to create a Multisig account, fund it and execute a transaction from this newly created account to send some funds to another address.
```

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
```diff
@@ -0,0 +1,117 @@
+
+### Steps
+More specifically the code implements the following steps :
+1. Creates a new Multisig account from the addresses of :
```
Suggestion to mention here that these 3 are the signatories from this point on.

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
```diff
@@ -0,0 +1,117 @@
+
+## Setup environment
+
+1) Fetch the latest Substrate or Polkadot/Kusama node from the above link. Follow instructions to build it, and start a dev chain.
```
```suggestion
1) Fetch the latest Substrate or Polkadot/Kusama node from the above link. Follow instructions to build it, and start a dev chain :
```

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
````diff
@@ -0,0 +1,117 @@
+    target/release/polkadot --dev
+    ```
+
+2) Install dependencies and build the JS target
````
```suggestion
2) Install dependencies and build the JS target :
```

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
````diff
@@ -0,0 +1,117 @@
+    yarn run build
+    ```
+
+3) Run the example script (see [src/multisig.ts](src/multisig.ts)). It will interact with your local node.
````
```suggestion
3) Run the example script (see [src/multisig.ts](src/multisig.ts)). It will interact with your local node :
```

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
```diff
@@ -0,0 +1,117 @@
+2. Accordingly the `asMulti` call is signed by Bob so the array passed in `otherSignatories` has to exclude Bob's address (`otherSignatoriesSortedExBob`)
+
+### Timepoint of `approveAsMulti` & Storage
+- In the `maybeTimepoint` parameter of the `asMulti` call we need to pass the block height and index of the `approveAsMulti` call. In order to to retrieve that dynamically we need to :
```
```suggestion
- In the `maybeTimepoint` parameter of the `asMulti` call we need to pass the block height and index of the `approveAsMulti` call. In order to to retrieve this information from the node chain storage we need to :
```

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
```diff
@@ -0,0 +1,117 @@
+### Timepoint of `approveAsMulti` & Storage
+- In the `maybeTimepoint` parameter of the `asMulti` call we need to pass the block height and index of the `approveAsMulti` call. In order to to retrieve that dynamically we need to :
+  1. Create the Storage key of our Multisig Storage item
+  2. Make an RPC request with the `state_getStorage` endpoint (using the Storage key from step 1 to retrieve the Multisig storage data from the chain. This data is retrieved as a SCALE-encoded byte array.
```
```suggestion
  2. Make an RPC request with the `state_getStorage` endpoint (using the Storage key from step 1 to retrieve the relevant Multisig storage data from the chain. This data is retrieved as a SCALE-encoded byte array.
```

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
````diff
@@ -0,0 +1,117 @@
+	
+      ```bash
+      Twox128("Multisig") + Twox128("Multisigs") + Twox64(multisigAddress) + multisigAddress + Blake128(multisigCallHash) + multisigCallHash
+      ```
````
I would add a note here explaining that we know that this is the right schema by consulting the source code for the pallet storage (in this specific case, this is at line 161  in   https://github.com/paritytech/substrate/blob/7d233c2446b5a60662400a0a4bcfb78bb3b79ff7/frame/multisig/src/lib.rs#L161

Imod7 replied:
Added it by referring to the `Multisigs` storage item under the `Multisig` pallet. Omitted the link in case the code changes but thank you for adding it here so I could double check that we are referring to the same item.

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
```diff
@@ -0,0 +1,117 @@
+NOTE: The example is working without this step since we are creating the multisig account also from the code.
+
+
+## Key Takeways from the Multisig example
```
```suggestion
## Key Takeaways 
```

IkerAlus commented on packages/txwrapper-examples/docs/MULTISIG.md:
```diff
@@ -0,0 +1,117 @@
+- Select the accounts of Alice, Bob And Charlie from the list as the signatories
+- Choose a name for your Multisig account, eg. `multi`
+- Click the "Create" button
+NOTE: The example is working without this step since we are creating the multisig account also from the code.
```
```suggestion
NOTE: The example code already creates the multisig account  so this step is not considered.
```

IkerAlus reviewed (changes requested):
I added some suggestions, most of them minor, to improve the explanations. Apart from that, I think the guide is on point and the added takeaways are very helpful.

Imod7 committed:
Update the description

Co-authored-by: IkerAlus <34474035+IkerAlus@users.noreply.github.com>

Imod7 committed:
Updated the md file with Iker's feedback

Imod7 committed:
Restructuring
- Restructured the folders so that each example is in a separate folder, inside of which there are files and subfolders (eg. 'src' and 'images') related to the specific example.
- Moved file 'util.ts' in the newly created folder 'common' since it is used in multiple examples.
- Updated the path of 'util.ts' in the example source files.
- Created a general README.md inside 'txwrapper-examples' folder that should give a short description about all examples and the folder structure.
- Updated the examples paths in “tsconfig.build.json” and “package.json” files.

Imod7 committed:
Run yarn lint --fix

IkerAlus reviewed (approved):
Looks all good to me

Imod7 committed:
Updated the general README & the polkadotBatchAll file

jsdw commented on packages/txwrapper-examples/README.md:
```diff
@@ -1,76 +1,3 @@
-## Questions?
-
-Open up a github issue [here](https://github.com/paritytech/txwrapper-core/issues)
+## Description
```
Perhaps this readme should have pointers to the examples, or some brief summary of the package, now?

jsdw replied:
Ah you are clearly ahead of me!

Imod7 replied:
Yes! Thank you! It was in my **Pendings** list in the PR description. 
- Since a lot of information now is in each example's folder and its `README` file, I wrote something very brief in the general `README` to describe what is included in the current folder. 
- Also, I updated the `README` of polkadotBatchAll example.

@jsdw Wdyt ? Should I add more details in any of these 2 files ?

jsdw replied:
The example code looks clear and well commented to me, so I don't think anything needs adding to the readme! lgtm! (better to avoid duplicating this stuff imo so that there is less stuff to keep in sync :))

TarikGul commented on packages/txwrapper-examples/package.json:
```diff
@@ -14,10 +14,10 @@
-    "polkadotBatchAll": "node lib/polkadotBatchAll",
+    "polkadot": "node lib/polkadot/src/polkadot",
+    "polkadotBatchAll": "node lib/polkadotBatchAll/src/polkadotBatchAll",
     "mandala": "node lib/mandala",
```
Feel free to remove `"mandala": "node lib/mandala"`

TarikGul reviewed (approved):
Great Job!
